### ISSUE TITLE
minetest.raycast seems to not contain players

### ISSUE THREAD

**Issue by adrido** [minetest/minetest#6157]
I try to use minetest.raycast for my cannons mod and it works great for nodes and objects, but it does not hit any player. here is a codesample to reproduce:
```lua
minetest.register_chatcommand("testray", {
	params = "",
	description = "Start a raycast and try to hit the player",
	func = function(name, param)
		local player = minetest.get_player_by_name(name)
		local pos = player:getpos()
		local ray = minetest.raycast({x=pos.x+1, y=pos.y+0.5, z=pos.z+1}, {x=pos.x-1, y=pos.y+0.5, z=pos.z-1}, true, true)
		for pointed_thing in ray do
			minetest.chat_send_player(name, "pointed_thing: "..dump(pointed_thing))
		end
		return true, "Done."
	end,
})
```

**Comment by SmallJoker** [comment]
@juhdanad Could this be the cause?
ServerEnvironment: https://github.com/minetest/minetest/blob/master/src/serverenvironment.cpp#L1680
ClientEnvironment: https://github.com/minetest/minetest/blob/master/src/clientenvironment.cpp#L621
`getBasePos` vs `getPos`

**Comment by stujones11** [comment]
Does #6007 make any difference, perhaps this is due to the offset?

**Comment by adrido** [comment]
No #6007 did not solve it. Trying the chat command from above with a y-offsets of +0; +0.1; +0.5; +0.9; +1 do not help.

**Comment by stujones11** [comment]
Have you tested this in multiplayer mode by casting a ray though a player other than yourself? I am just wondering if perhaps the client does 'see' its own player selection box.

**Comment by adrido** [comment]
Nope, sadly doesn't work in multiplayer.
Updated testscript: 

```Lua
minetest.register_chatcommand("testray", {
	params = "[playername]",
	description = "testray: start a raycast and try to hit the player",
	func = function(name, param)
		if param ~= "" then 
			name = param
		end
		local player = minetest.get_player_by_name(name)
		if not player then
			return false, "Player "..name.."' not found!"
		end
		local pos = player:getpos()
		local ray = minetest.raycast({x=pos.x+1, y=pos.y+0.5, z=pos.z+1}, {x=pos.x-1, y=pos.y+0.5, z=pos.z-1}, true, true)
		for pointed_thing in ray do
			minetest.chat_send_player(name, "pointed_thing: "..dump(pointed_thing))
		end

		return true, "Done."
	end,
})
```

**Comment by stujones11** [comment]
@adrido can you confifim that #6241 fixes this issue?

**Comment by adrido** [comment]
yes it does! 👍 

Thanks for your work.

**Comment by adrido** [comment]
Why got this closed? #6241 is still open. 😕

**Comment by stujones11** [comment]
> Thanks for your work.

No problem, I need this working for my shooter mod as well ;-) No idea why it's been closed though, I guess @rubenwardy thought #6241 was merged.
